Source/Core/AST/Absy.cs

```diff
@@ -280,6 +280,33 @@ public List<int> FindLayers()
       }
       return layers.Distinct().OrderBy(l => l).ToList();
     }
+
+    // Look for {:name string} in list of attributes.
+    public string FindStringAttribute(string name)
+    {
+      return QKeyValue.FindStringAttribute(Attributes, name);
+    }
+
+    public void AddStringAttribute(IToken tok, string name, string parameter)
+    {
+      Attributes = new QKeyValue(tok, name, new List<object>() {parameter}, Attributes);
+    }
+
+    public void CopyIdFrom(IToken tok, ICarriesAttributes src)
+    {
+      var id = src.FindStringAttribute("id");
+      if (id is not null) {
+        AddStringAttribute(tok, "id", id);
+      }
+    }
+
+    public void CopyIdWithSuffixFrom(IToken tok, ICarriesAttributes src, string suffix)
```

Comment:
`target.Copy` reads strangely. I would expect the `src` to be on the left side of the copy verb